news-bundle/src/Resources/contao/classes/News.php

```diff
@@ -577,19 +577,23 @@ private function getPageWithDetails($intPageId)
 	{
 		if (!isset(self::$arrPageCache[$intPageId]))
 		{
-			$objPage = PageModel::findWithDetails($intPageId);
+			$objPage = self::$arrPageCache[$intPageId] = PageModel::findWithDetails($intPageId);
+
+			if (null === $objPage)
+			{
+				return null;
+			}
+
 			$objLayout = LayoutModel::findByPk($objPage->layout);
 
 			if (null === $objLayout)
 			{
-				return null;
+				return (self::$arrPageCache[$intPageId] = null);
```

Comment:
This does not have any effect at all, because we are using `!isset(self::$arrPageCache[$intPageId])` in line 578. Please only add the early return statements and leave the rest as is.

Reply:
We need to set `self::$arrPageCache[$intPageId]` to `null` here, otherwise `$objLayout = LayoutModel::findByPk($objPage->layout);` will get executed for the same page every time subsequently.

Reply:
`!isset()` will return `false` if the value is `null`, therefore, as I said above, your changes do not have any effect. 🤷‍♂️

Reply:
Ah yes, then we need `array_key_exists`.